src/effects/lv2/lv2effectprocessor.cpp

```diff
@@ -0,0 +1,102 @@
+#include "effects/lv2/lv2effectprocessor.h"
+#include "engine/effects/engineeffect.h"
+#include "control/controlobject.h"
+#include "util/sample.h"
+
+#define MAX_BUFFER_LEN 160000
+
+LV2EffectProcessor::LV2EffectProcessor(EngineEffect* pEngineEffect,
+                                       EffectManifestPointer pManifest,
+                                       const LilvPlugin* plugin,
+                                       QList<int> audioPortIndices,
+                                       QList<int> controlPortIndices) {
+    m_sampleRate = ControlObject::get(ConfigKey("[Master]", "samplerate"));
+ 
+    m_handle = lilv_plugin_instantiate(plugin, m_sampleRate, NULL); 
+    if (!m_handle) {
+        return;
+    } 
+
+    m_inputL = new float[MAX_BUFFER_LEN];
+    m_inputR = new float[MAX_BUFFER_LEN];
+    m_outputL = new float[MAX_BUFFER_LEN];
+    m_outputR = new float[MAX_BUFFER_LEN];
+    m_params = new float[pManifest->parameters().size()];
+
+
+    const QList<EffectManifestParameterPointer>& effectManifestParameterList =
+            pManifest->parameters();
+
+    // Initialize EngineEffectParameters
+    for (const auto& pParam: effectManifestParameterList) {
+        m_parameters.append(pEngineEffect->getParameterById(pParam->id()));
+    }
+
+    for (int i = 0; i < m_parameters.size(); i++) {
+        m_params[i] = m_parameters[i]->value();
+        lilv_instance_connect_port(m_handle, controlPortIndices[i], &m_params[i]);
+    }
+
+    // We assume the audio ports are in the following order:
+    // input_left, input_right, output_left, output_right
+    lilv_instance_connect_port(m_handle, audioPortIndices[0], m_inputL);
+    lilv_instance_connect_port(m_handle, audioPortIndices[1], m_inputR);
+    lilv_instance_connect_port(m_handle, audioPortIndices[2], m_outputL);
+    lilv_instance_connect_port(m_handle, audioPortIndices[3], m_outputR);
+
+    lilv_instance_activate(m_handle);
```

Comment:
This is not the correct place to call this function. The LV2 plugin's `activate()` function needs to be called before `LV2EffectProcessor::process` whenever the effect is enabled. We could hack this into the existing `LV2EffectProcessor::process` function by using the EnableState like native effects, but I think it would be a better idea to add an `activate()` method to the `EffectProcessor` API and refactor the native effects to use it. Then we might be able to get rid of the intermediate ENABLING and DISABLING EffectProcessor::EnableStates.